crates/sui-framework/sources/governance/stake_subsidy.move

```diff
@@ -23,44 +25,34 @@ module sui::stake_subsidy {
     const STAKE_SUBSIDY_DECREASE_RATE: u128 = 1000; // in basis point
     const STAKE_SUBSIDY_PERIOD_LENGTH: u64 = 30; // in number of epochs
 
-    public(friend) fun create(initial_stake_subsidy_amount: u64): StakeSubsidy {
+    public(friend) fun create(balance: Balance<SUI>, initial_stake_subsidy_amount: u64): StakeSubsidy {
         StakeSubsidy {
             epoch_counter: 0,
-            balance: balance::zero(),
+            balance,
             current_epoch_amount: initial_stake_subsidy_amount,
         }
     }
 
-    public(friend) fun mint_stake_subsidy_proportional_to_total_stake_testnet(
-        subsidy: &mut StakeSubsidy, supply: &mut Supply<SUI>, stake_subsidy_rate: u64, total_stake: u64
-    ) {
-        let amount_to_mint = ((total_stake as u128) * (stake_subsidy_rate as u128)) / BASIS_POINT_DENOMINATOR;
-        balance::join(
-            &mut subsidy.balance, 
-            balance::increase_supply(supply, (amount_to_mint as u64))
-        );
-    }
+    /// Advance the epoch counter and draw down the subsidy for the epoch.
+    public(friend) fun advance_epoch(subsidy: &mut StakeSubsidy): Balance<SUI> {
+        // Take the minimum of the reward amount and the remaining balance in
+        // order to ensure we don't overdraft the remaining stake subsidy
+        // balance
```

Comment:
super mega nit that you should feel fine ignoring if you want: could `balance` live on the line above (we aren't really keeping to 80 char limits), and could you add an ending period?